src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs

```diff
@@ -282,14 +284,22 @@ public async Task ProcessFinishedReleasesAsync(CancellationToken cancellationTok
                         }
                         else
                         {
-                            Logger.LogInformation($"All releases for build {buildId} for channel {channelId} finished. Creating BuildChannel.");
+                            if (successfulRelease)
+                            {
+                                Logger.LogInformation($"All releases for build {buildId} for channel {channelId} finished. Creating BuildChannel.");
 
-                            buildChannelsToAdd.Add(new BuildChannel
+                                buildChannelsToAdd.Add(new BuildChannel
+                                {
+                                    BuildId = buildId,
+                                    ChannelId = channelId
+                                });
+                                await runningPipelines.TryRemoveAsync(tx, buildId);
```

Comment:
Shouldn't we remove it from the running releases collection anyways? otherwise we are going to keep logging the same error over and over for a failed release.

Reply:
Sadly that depends. Let's say that one of the steps in the pipeline failed due to a network failure. A retry could potentially make the release work. I don't think that case should be addressed here though. We have an issue for supporting retries: https://github.com/dotnet/arcade/issues/2769

Reply:
So, to be clear. I definitely agree with you. We should 1) remove the release from the running releases collection, 2) not assign the release to a channel and 3) create a GH issue for it.

Reply:
But this won't retry yet, this will just keep logging errors every 5 minutes because it will keep seeing the release failed.
If the idea that someone is going to manually go and retry it, and then this service will look at it and add the build to the channel, I'd rather have a separate collection for those releases that get checked on a separate cadence and don't log this much.

Reply:
If we create a GH issue we are removing the release from the collection (see [this](https://github.com/dotnet/arcade-services/pull/420/files/182c686358d692541c28ae6106e2c96f80e86d1b#diff-a2c05c8a9be8cf4b350cddf00dee681bR265))

Reply:
As @riarenas recommends, should we do `await runningPipelines.TryRemoveAsync(tx, buildId);` on either successful or unsuccessful case?

Reply:
I think you shouldn't do `await runningPipelines.TryRemoveAsync(tx, buildId);` when only a single release failed for the build, otherwise you'll lose track of other potential releases running for that build. Still, if at least one of the releases for a build+channel failed the build shouldn't be assigned to that channel. Do you agree @mmitche ?